Feature/add sign in uri parameter

sria-shoji opened the pull request:
Thank you for providing a great SDK.
The purpose of this change is to make Cognito's authentication page customizable.
In the current source code, the URL of the authentication page is generated based on the parameter "CognitoAuthWebDomain" in the plist file.
I would like to create an authentication page in a domain different from "CognitoAuthWebDomain" and display it in the in-app browser. For that purpose, we added a new parameter "CognitoAuthSignInUri" to the plist file. By setting the URL of the authentication page to this parameter, the custom authentication page is displayed.
Since I need this change, I would appreciate it if you merge the pull request. Thank you for your consideration.

minbi commented:
Hi @sria-shoji ,

Thanks for the pull request. The change as it is cannot be accepted because it would break existing customers by changing the method signature.

If I understand correctly, this change will support custom domains that do not follow the `<webdomain>/oauth2/authorize` which means `<webdomain>/otherpath` will also work after hte change?

You will need to create a separate init method without the `webDomain` variable to support this case.

sria-shoji commented:
Hi @minbi,
Thank you for confirming the changes and comments.

As you say, given the impact on other customers, changes to the init method were incorrect.

> If I understand correctly, this change will support custom domains that do not follow the <webdomain> / oauth 2 / authorize which means <webdomain> / otherpath will also work after hte change?

I think so.
However, as before, <webdomain> will be used for getting the token.
The purpose is to display a page of custom URL which is not "<webdomain> / oauth2 / authorize" when signing in.

I fixed the following points in the source code:
 1. I replaced the signatures of the two init methods that affect others. However, these two methods call the newly added init method.
 2. I added a new init method so that it accepts arguments of <signInUri>.

This will work as before if there is no <CognitoAuthSignInUri> key in info.plist, and if there is a key, use the value of this key to display the ViewController for signing in.

Could you please accept this?

sria-shoji commented:
How about this matter?
Thank you for your answer.
